README.md

```diff
@@ -2,16 +2,171 @@
+
+### Configuring the uninstall Maven goal
+
+A sample pom.xml file configuration of the uninstall Maven goal is shown below.
+
+      <build>
+    	<plugins>
+        	<plugin>
+            	<groupId>org.wso2.carbon.maven</groupId>
+            	<artifactId>carbon-feature-plugin</artifactId>
+            	<version>${carbon.feature.plugin.version}</version>
+            	<executions>
+                	<execution>
+                    	<id>p2-feature-generation</id>
+                    	<phase>package</phase>
+                    	<goals>
+                        	<goal>uninstall</goal>
+                    	</goals>
+                    	<configuration>
+                        	//plugin configuration goes here.
+                    	</configuration>
+                	</execution>
+            	</executions>
+        	</plugin>
+            </plugins>
+          </build>
+	  
+You can modify the above file to add the configurations of the plugin by adding the following parameters within the `<configuration>` element of it. 
+
+* `destination`: Points to the `<CARBON_HOME>/repository/components/` directory.
+
+ MANDATORY property.
+ Example: `<destination>/home/Carbon/wso2carbon-4.4.0/repository/components</destination>`
+ 
+* `runtime`: Which runtime in the destination from where you need to uninstall the features. 
+
+ MANDATORY property.
+ Example: `<runtime>default</runtime>`.
+ 
+* `features`: List of features to be uninstalled from the destination runtime.
+
+ MANDATORY
+ Example:
+ 
+       <features>
+             <feature>
+                <id>org.wso2.carbon.registry.contentsearch.feature.group</id>
+                <version>4.4.0</version>
+             </feature>
+             <feature>
+                <id>org.wso2.ciphertool.feature.group</id>
+                <version>4.4.0</version>
+             </feature>
+        </features>
+        
+
+## P2 Touchpoint
+### Equinox P2 Touchpoint
+Features can have configuration files which need to be copied to a particular location when install the feature. This can be done by specifying the location in the p2.inf in the feature. For example:
+Example:
+instructions.configure = \
+org.eclipse.equinox.p2.touchpoint.natives.copy(source:${installFolder}/../lib/features/org.wso2.carbon.kernel_${feature.version}/launcher/,target:${installFolder}/../../bin/bootstrap/,overwrite:true);\
+
+All equinox p2 touchpoint are available at [Provisioning Actions and Touchpoints](http://help.eclipse.org/neon/index.jsp?topic=/org.eclipse.platform.doc.isv/guide/p2_actions_touchpoints.html)
+
+### Copy to specific runtime
+In C5, there are runtimes which are specific to each product and the feature developer will not know this. In order to support dynamically copying files to the runtime location during feature installation, WSO2 has introduced a custom touchpoint. 
+
+In the p2.inf, the following should be added in-order to copy a particular file to a runtime location:
```

Comment:
*The following should be added to the p2.inf in order to copy a particular file to a runtime location: